src/main/java/com/arpnetworking/utility/RegexAndMapReplacer.java

```diff
@@ -0,0 +1,296 @@
+            output.append(getReplacement(matcher, tokenBuilder.toString(), tokenNumeric, variables, variablesUsedBuilder));
+        } else {
+            // Consume until we hit a non-digit character
+            while (x < replacement.length()) {
+                c = replacement.charAt(x);
+                if (Character.isDigit(c)) {
+                    tokenBuilder.append(c);
+                } else {
+                    break;
+                }
+                x++;
+            }
+            if (tokenBuilder.length() == 0) {
+                throw new IllegalArgumentException(
+                        String.format(
+                                "Invalid replacement token, non-numeric tokens must be surrounded by { } at col %d: %s",
+                                x,
+                                replacement));
+            }
+            output.append(getReplacement(matcher, tokenBuilder.toString(), true, variables, variablesUsedBuilder));
+        }
+        return x - offset - 1;
+    }
+
+    private static String getReplacement(
+            final Matcher matcher,
+            final String replaceToken,
+            final boolean numeric,
+            final LinkedHashMap<String, Map<String, String>> variables,
+            final ImmutableList.Builder<String> variablesUsedBuilder) {
+        if (numeric) {
+            final int replaceGroup = Integer.parseInt(replaceToken);
+            return matcher.group(replaceGroup);
+        } else {
+            final int prefixSeparatorIndex = replaceToken.indexOf(':');
+            final String prefix;
+            final String variableName;
+            if (prefixSeparatorIndex == replaceToken.length() - 1) {
+                throw new IllegalArgumentException(
+                        String.format("found prefix in variable replacement, but no variable name found: '%s'",
+                        replaceToken));
+            }
+            if (prefixSeparatorIndex != -1) {
+                prefix = replaceToken.substring(0, prefixSeparatorIndex);
+                variableName = replaceToken.substring(prefixSeparatorIndex + 1);
+            } else {
+                prefix = "";
+                variableName = replaceToken;
+            }
+            if (prefix.isEmpty()) {
+                return replaceNonPrefix(matcher, variables, variablesUsedBuilder, variableName);
+
+            } else {
+                return replacePrefix(matcher, variables, variablesUsedBuilder, prefix, variableName);
+            }
+        }
+    }
+
+    private static String replacePrefix(
+            final Matcher matcher,
+            final LinkedHashMap<String, Map<String, String>> variables,
+            final ImmutableList.Builder<String> variablesUsedBuilder,
+            final String prefix,
+            final String variableName) {
+        if (prefix.equals("capture")) {
+            if (isNumeric(variableName)) {
+                final Integer groupIndex = Integer.valueOf(variableName);
```

Comment:
Same comment about about numeric being something other than a capture group index and also the function calling this already special cased this.

Reply:
^ This can also throw or be invalid if it overflows.

Reply:
Fixed.